avocado_vt/test.py

```diff
@@ -221,8 +226,13 @@ def __safe_env_save(self, env):
             env.save()
         except Exception as details:
             if hasattr(stacktrace, "str_unpickable_object"):
-                self.log.warn("Unable to save environment: %s",
-                              stacktrace.str_unpickable_object(env.data))
+                try:
+                    self.log.warn("Unable to save environment: %s",
+                                  stacktrace.str_unpickable_object(env.data))
+                except ValueError:
+                    self.log.warn("Unable to save environment, even though "
+                                  "it's pickable: %s (%s)", details,
+                                  env.data)
```

Comment:
Why don't stop overriding self.__param at once? I mean:

```diff
diff --git a/avocado_vt/test.py b/avocado_vt/test.py
index 4ff57d23..ee379361 100644
--- a/avocado_vt/test.py
+++ b/avocado_vt/test.py
@@ -92,7 +92,7 @@ class VirtTest(test.Test):
         :param vt_params: avocado-vt/cartesian_config params stored as
                           `self.params`.
         """
-        self.__params = None
+        self.__params_vt = None
         self.__avocado_params = None
         self.bindir = data_dir.get_root_dir()
         self.virtdir = os.path.join(self.bindir, 'shared')
@@ -116,7 +116,7 @@ class VirtTest(test.Test):
             # 36LTS set's `self.params` instead of having it as a property
             # which stores the avocado params in `self.__params`
             self.__avocado_params = self.__params
-        self.__params = utils_params.Params(vt_params)
+        self.__params_vt = utils_params.Params(vt_params)
         self.debugdir = self.logdir
         self.resultsdir = self.logdir
         self.timeout = vt_params.get("test_timeout", self.timeout)
@@ -132,8 +132,8 @@ class VirtTest(test.Test):
         once the Avocado-vt params are set it reports those instead. This
         is necessary to complete the `avocado.Test.__init__` phase
         """
-        if self.__params is not None:
-            return self.__params
+        if self.__params_vt is not None:
+            return self.__params_vt
         else:
             # The `self.__params` is set after the `avocado.test.__init__`,
             # but in newer Avocado `self.params` is used during `__init__`
@@ -145,7 +145,7 @@ class VirtTest(test.Test):
         """
         For compatibility with 36lts we need to support setter on params
         """
-        self.__params = value
+        self.__params_vt = value
 
     @property
     def avocado_params(self):
@@ -175,15 +175,6 @@ class VirtTest(test.Test):
         """
         return None
 
-    def get_state(self):
-        """
-        Avocado-vt replaces Test.params with avocado-vt params. This function
-        reports the original params on `get_state` call.
-        """
-        state = super(VirtTest, self).get_state()
-        state["params"] = self.avocado_params
-        return state
-
     def _start_logging(self):
         super(VirtTest, self)._start_logging()
         root_logger = logging.getLogger()

```

Then all Avocado will need is:


```diff
diff --git a/avocado/core/test.py b/avocado/core/test.py
index 4ab33b43..d9def132 100644
--- a/avocado/core/test.py
+++ b/avocado/core/test.py
@@ -644,7 +644,7 @@ class Test(unittest.TestCase, TestData):
         state['job_unique_id'] = self.job.unique_id
         state['params'] = [(path, key, value)
                            for path, key, value
-                           in self.params.iteritems()]
+                           in self.__params.iteritems()]
         return state
 
     def _register_log_file_handler(self, logger, formatter, filename,

```

Reply:
Yep, that's a good idea. I'll send v2 later today (need to finish something).